doc/source/dev/toolchain.rst

```diff
@@ -144,69 +145,49 @@ SciPy has been restricted in the use of more advanced language features by the
+================  =======================================================================
 
 
-C and C++ Language Standards
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+C++ Language Standards
+~~~~~~~~~~~~~~~~~~~~~~
 
-C and C++ language standards for SciPy are generally guidelines
+C++ language standards for SciPy are generally guidelines
 rather than official decisions. This is particularly true of
 attempting to predict adoption timelines for newer standards.
 
 ================  =======================================================================
- Date              C/C++ Standard
+ Date              C++ Standard
 ================  =======================================================================
- <= 2018           C90
- 2019              C90 for old code, may consider C99 for new
- 2020              C99
+ <= 2019           C++03
  2020              C++11
  2021              C++14
- ?                 C11, C17, C++17, C++20
+ ?                 C++17, C++20, C++23
 ================  =======================================================================
 
-For C, C11/C17 support will be available as soon as the ``vmImage`` for
-building SciPy is upgraded to ``windows-2019`` (which is compatible with
-currently supported CPython versions and "just" needs to be executed). This is
-because GCC & LLVM support all relevant C11 features with the oldest currently
-used versions, and C17 is just a bugfix for C11, as mentioned above.
-
-On the C++ side, since dropping support for Python 2.7, C++11 can be used
+Since dropping support for Python 2.7, C++11 can be used
 universally. For C++14, Windows is not a restriction anymore since Visual
 Studio 15.9 (<-> _MSC_VER 19.16, see [8]_), has full support (same for C++17),
-see [4]_. However, using C++14 still requires bumping the GCC minimal
-requirement to 5.x and C++17 will require GCC >= 7 [4]_.
-Compiler support for C++20 is still under heavy development.
-
-.. note::
-
-    Developer Note: Some C99 features would be useful for scientific
-    programming, in particular better support of IEEE 754 [5]_.
-    SciPy has a small include file ``scipy/_lib/_c99compat.h`` which
-    provides access to a few functions. Use in conjunction
-    with ``<numpy/npy_math.h>``.
-
-    ========================================= ========================================================
-     Feature                                  Workaround
-    ========================================= ========================================================
-    ``isnan()``, ``isinf()``, ``isfinite()``  Use ``sc_isnan()``, ``sc_isinf()``, ``sc_isfinite()``
-    ``NAN``                                   Use ``NPY_NAN`` (it is *almost* equivalent)
-    inline functions                          Make static functions and place in an include .h file
-    mid-block variable declarations           Declare variables at the top of the block
-    ========================================= ========================================================
```

Comment:
I believe all of this is outdated and can now be used unconditionally, but will happily stand corrected if I'm wrong about that (or if `scipy/_lib/_c99compat.h` should still be mentioned, if only as slated for removal).

In the meantime, I've repurposed link Nr. 5 to link to MSVC implementation of `<complex.h>` which I saw [mentioned](https://github.com/numpy/numpy/issues/18378#issuecomment-785993494) in an issue that's targetted for 1.8 (so sadly, that's not yet in the card).

Reply:
Agreed, we can (and do) require C99 now.